server/text_generation_server/utils/layers.py

```diff
@@ -369,7 +369,7 @@ def forward(self, hidden_states, residual=None):
     import rotary_emb
 
     class PositionRotaryEmbedding(nn.Module):
-        def __init__(self, inv_freq):
+        def __init__(self, inv_freq, scale_factor=1, dynamic_scaling=False, max_seq_len=2048, dim=None, base=None):
```

Comment:
Can we have at most 1 extra argument.

A lot of information should be extractable directly from `inv_freq`.

Reply:
Yup, I can try to simplify this